apps/backend/src/groups/groups.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Group, GroupMembers, Prisma, Role, User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+@Injectable()
+export class GroupsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(data: Prisma.GroupCreateInput): Promise<Group> {
+    return await this.prisma.group.create({ data });
+  }
+
+  async findAll(): Promise<Group[]> {
+    if (!this.prisma.group) {
+      throw new NotFoundException('Groups not found');
+    }
+    return await this.prisma.group.findMany();
+  }
+
+  async findOne(id: number): Promise<Group> {
+    const group = await this.prisma.group.findUnique({ where: { id: id } });
+    if (!group) {
+      throw new NotFoundException(`Group with ID ${id} not found`);
+    }
+    return group;
+  }
+
+  async findMembers(id: number): Promise<User[]> {
+    const userids = await this.prisma.groupMembers.findMany({
+      where: {
+        groupId: id,
+      },
+    });
+    const members = await this.prisma.user.findMany({
+      where: {
+        id: {
+          in: userids.map((member) => member.userId),
+        },
+      },
+    });
+    return members;
+  }
```

Comment:
this can be simplified with nest includes
https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries#include-a-relation

Might need some transformation on the data once it's returned from the DB, but it can be done in one query, and that's usually preferred